test/unit_test/interactions_test.cc

```diff
@@ -27,6 +27,28 @@ struct eval_gen_data
   }
 };
 
```

Comment:
based on a comment from `interaction_term.cc` if it is legit to compare char's with full namespace's then can we add a test here for that too?

Reply:
I haven't implemented hash based expansion (next PR). I will add a test which sorts interactions with both character and hash based interactions in that PR.